misk-metrics-testing/src/main/kotlin/misk/metrics/FakeMetrics.kt

```diff
@@ -32,7 +31,8 @@ class FakeMetrics @Inject internal constructor(
     name: String,
     help: String,
     labelNames: List<String>,
-    quantiles: Map<Double, Double>
+    quantiles: Map<Double, Double>,
+    maybeMaxAgeSeconds: Long?
```

Comment:
nit: I'd keep this (and the other instances of it) `maxAgeSeconds: Long?`, maybe with a default.

Reply:
+1 for default

Reply:
Thanks for the feedback.

> I'd keep this (and the other instances of it) maxAgeSeconds: Long?

I was trying to avoid the code below being confusing.

```
  if (maxAgeSeconds != null) {
    maxAgeSeconds(maxAgeSeconds)
  }
```

Where `maxAgeSeconds` represents both a variable and a function on the summary builder. I guess the alternative is to use `this` to make it clear which one is a function.

> adding a default value

I am a little confused about the right use of defaults in Kotlin. There is a default on the interface:

https://github.com/cashapp/misk/pull/2271/files#diff-4a56fc32fc0a0bfbf879f1050519633c6d3d4c97afcebcddc3b4f2e74087d7e9L65-R66

This class implements that interface. Do I also need to add the default here or does it inherit the default?

If I try adding a default to this function, I get an error message:

> An overriding function is not allowed to specify default values for its parameters

Reply:
You're correct, the default should be defined in the interface (and gets inherited)